kratos/spatial_containers/geometrical_objects_bins.h

```diff
@@ -95,166 +107,219 @@ class GeometricalObjectsBins
+        }
+    }
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it in a given radius.
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * If there are no objects in that radius the result will be set to not found.
+     * Result contains a flag is the object has been found or not.
+     * @param rPoint The point to be checked
+     * @param Radius The radius to be checked
+     * @return ResultType The result of the search
+     */
+    ResultType SearchNearestInRadius(
+        const Point& rPoint,
+        const double Radius
+        );
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it in a given radius (iterative version).
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * If there are no objects in that radius the result will be set to not found.
+     * Result contains a flag is the object has been found or not.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param Radius The radius to be checked
+     * @return std::vector<ResultType> The result of the search
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<typename TPointIteratorType>
+    std::vector<ResultType> IterativeSearchNearestInRadius(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        const double Radius
+        )
+    {
+        // Doing a vector of results
+        std::vector<ResultType> results;
+        const std::size_t number_of_points = std::distance(itPointBegin, itPointEnd);
+        results.resize(number_of_points);
+        for (auto it_point = itPointBegin ; it_point != itPointEnd ; it_point++){
+            results[it_point - itPointBegin] = SearchNearestInRadius(*it_point, Radius);
+        }
+        return results;
+    }
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it.
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * Result contains a flag is the object has been found or not.
+     * @param rPoint The point to be checked
+     * @return ResultType The result of the search
+    */
+    ResultType SearchNearest(const Point& rPoint);
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it (iterative version).
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * Result contains a flag is the object has been found or not.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @return std::vector<ResultType> The result of the search
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<typename TPointIteratorType>
+    std::vector<ResultType> IterativeSearchNearest(
```

Comment:
I think the name of this function is quite misleading... In fact, this seems a little weird to me, what is the purpose of this function?

Reply:
I will admit that it was the only way to avoid exposing manually methods in python.  I will rename and expose the methods manually in python. Sorry for the noise.